File: Competitive Programming 3/Section 1.3.3/661 - Blowing Fuses/Python/main.py
def main():
    cs = 1 
    while True:
        n,m,c=map(int,input().split())
        if n == 0 and m==0 and c ==0:
            break
        print("Sequence "+str(cs))
        cs+=1
        isTurnedOn,consumption = [False] * 22,[0] * 22
        for i in range(1,n+1):
            consumption[i]=int(input())
        consumed,maxConsumed=0,0
        for i in range(0,m):
            toggleI = int(input())
            if isTurnedOn[toggleI] == True:
                consumed-=consumption[toggleI]
            else:
                consumed+=consumption[toggleI]
            isTurnedOn[toggleI]= not isTurnedOn[toggleI]
            maxConsumed = max(maxConsumed,consumed)
        print("Fuse was "+("blown" if maxConsumed > c else "not blown")+".")
        if maxConsumed <= c:
            print("Maximal power consumption was "+str(maxConsumed)+" amperes.")
        print()
main()

File: Competitive Programming 3/Section 1.3.3/661 - Blowing Fuses/C++/main.cpp
#include<bits/stdc++.h>

using namespace std;

bool isTurnedOn[22];
int consumption[22];
int n,m,c;

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    int cs=1;
    while(cin>>n>>m>>c,n||m||c){
        memset(isTurnedOn,0,sizeof(isTurnedOn));
        cout << "Sequence " << cs++<<"\n";
        for(int i = 1 ; i <= n ; i ++)
            cin >> consumption[i];
        int consumed = 0, maxConsumed=0;
        while(m--){
            int toggleI;
            cin >> toggleI;
            consumed+=consumption[toggleI]*(isTurnedOn[toggleI]?-1:1);
            isTurnedOn[toggleI]=!isTurnedOn[toggleI];
            maxConsumed=max(maxConsumed,consumed);
        }
        cout << "Fuse was " << (maxConsumed>c?"blown":"not blown") << ".\n";
        if(maxConsumed<=c)
            cout << "Maximal power consumption was " << maxConsumed << " amperes.\n";
        cout << "\n";
    }
    return 0;
}